FraudProofs.md

```diff
@@ -0,0 +1,154 @@
+# Fraud Proofs
+
+## Introduction
+
+In the Discreet Log Contract oracle model, oracles act as oblivious public entities while
+their users act privately.
+
+Due to DLC oracles' public nature, should they ever commit fraudulent acts such as spreading
+false, invalid, or conflicting attestations, their fraud should be provable and verifiable from public information.
+After all, oracle fraud comes in the form of an unforgeable attestation of false or invalid events,
+or else in the form of multiple conflicting unforgeable attestations.
+
+This document specifies a standard format for serializing fraud proofs as well as specifying their verification.
+
+## Table of Contents
+
+* [Proof of Oracle Attestation](#proof-of-oracle-attestation)
+  * [Version 0 `oracle_attestation_proof`](#version-0-oracle_attestation_proof)
+* [Claim of False Oracle Attestation](#claim-of-false-oracle-attestation)
+  * [Version 0 `false_oracle_attestation_claim` ](#version-0-false_oracle_attestation_claim)
+* [Proof of Invalid Oracle Attestation](#proof-of-invalid-oracle-attestation)
+  * [Version 0 `invalid_oracle_attestation_proof`](#version-0-invalid_oracle_attestation_proof)
+* [Proof of Oracle Equivocation](#proof-of-oracle-equivocation)
+  * [Version 0 `oracle_equivocation_proof`](#version-0-oracle_equivocation_proof)
+  * [Version 1 `oracle_equivocation_proof`](#version-1-oracle_equivocation_proof)
+* [Authors](#authors)
+
+## Proof of Oracle Attestation
+
+The fundamental building block of DLC oracle fraud proofs is a proof that a specific oracle
+attested to a specific outcome, which can then be used with other information to prove fraud.
+
+In theory, it would be best if DLC oracles could deliver their announcements and attestations
+using some form of broadcast channel, but in practice it is expected that oracle messages will
```

Comment:
> I would rather say anonymous network. Note, I'm not sure how you dissociate broadcast channel from P2P means in this sentence.

The distinction I'm trying to make is between a twitter-like model where you can't easily tweet and have it appear different to different people as opposed to a more P2P setting where it is easier to lie to certain kinds of nodes "in private" and the much more private interaction of a REST API.

I changed the wording a little to say "public broadcast channel" and "certain P2P means."

> Maybe say that p2p propagation of oracles events is left for future spec efforts ?

Doesn't really seem relevant here

> Also "that oracle messages will often times be received". Is that an english sentence ?

often times has been changed to oftentimes but otherwise, I think so?